packages/server/modules/automate/services/trigger.ts

```diff
@@ -177,6 +181,46 @@ const createAutomationRunData =
     return automationRun
   }
 
+export type SetFunctionRunStatusReportDeps = {
+  getAutomationFunctionRunRecord: typeof getFunctionRun
+  upsertAutomationFunctionRunRecord: typeof upsertAutomationFunctionRun
+}
+
+export const setFunctionRunStatusReport =
+  (deps: SetFunctionRunStatusReportDeps) =>
+  async (
+    params: Pick<
+      AutomationFunctionRunRecord,
+      'runId' | 'status' | 'statusMessage' | 'contextView' | 'results'
+    >
+  ): Promise<boolean> => {
+    const { getAutomationFunctionRunRecord, upsertAutomationFunctionRunRecord } = deps
+    const { runId, ...statusReportData } = params
+
+    const currentFunctionRunRecord = await getAutomationFunctionRunRecord(runId)
+
+    if (!currentFunctionRunRecord) {
+      return false
```

Comment:
this probably needs to be an error, resulting in a `FunctionRunNotFound` error case in graphql

Reply:
added, I think correctly? Do we need to report back the missing id in the message?